controllers/componentdetectionquery_controller.go

```diff
@@ -198,18 +206,9 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 					}
 				}
 			}
-
-			// Remove the cloned path if present
-			if isExist, _ := ioutils.IsExisting(r.AppFS, clonePath); isExist {
-				if err := r.AppFS.RemoveAll(clonePath); err != nil {
-					log.Error(err, fmt.Sprintf("Unable to remove the clonepath %s %v", clonePath, req.NamespacedName))
-					r.SetCompleteConditionAndUpdateCR(ctx, &componentDetectionQuery, err)
-					return ctrl.Result{}, nil
-				}
-			}
 		} else {
-			if componentDetectionQuery.Spec.IsMultiComponent {
-				errMsg := fmt.Sprintf("cannot set IsMultiComponent to %v and set DevfileURL to %s as well... %v", componentDetectionQuery.Spec.IsMultiComponent, source.DevfileURL, req.NamespacedName)
+			if isMultiComponent {
+				errMsg := fmt.Sprintf("Cannot set DevfileURL to %s for a multi component project... %v", source.DevfileURL, req.NamespacedName)
```

Comment:
I think this section needs to be handled differently now..

with spec.isMultiComponent gone, we cant throw this error anymore or handle this case like this anymore..

If someone is specifying `source.DevfileURL`, we just use this URL directly for devfile and dont even bother trying to read a devfile or analyzing using Alizer, so this should probably be the first step IMO

Reply:
So in summary if someone mentions spec.DevfileURL:

1. Use the devfile directly
2. Check if it has Dockerfile reference

we dont need to clone or use Alizer

Reply:
updated to skip detection when DevfileURL is set